freeipa/src/test/java/com/sequenceiq/freeipa/service/freeipa/flow/FreeIpaPostInstallServiceTest.java

```diff
@@ -70,11 +101,122 @@ public void postInstallFreeIpaExecutePostServiceDeploymentRecipes() throws Excep
         when(freeIpaRecipeService.hasRecipeType(1L, RecipeType.POST_SERVICE_DEPLOYMENT, RecipeType.POST_CLUSTER_INSTALL)).thenReturn(true);
         when(userSyncBindUserService.doesBindUserAndConfigAlreadyExist(stack)).thenReturn(false);
 
-        freeIpaPostInstallService.postInstallFreeIpa(1L, false);
+        underTest.postInstallFreeIpa(1L, false);
 
         verify(hostOrchestrator).postServiceDeploymentRecipes(eq(gatewayConfig), eq(nodes), any(StackBasedExitCriteriaModel.class));
         verify(userSyncBindUserService).doesBindUserAndConfigAlreadyExist(stack);
         verify(userSyncBindUserService).createUserAndLdapConfig(stack);
+        verifyNoInteractions(wiamClient, userSyncService);
+    }
+
+    @Test
+    public void testWiamCalledIfEntitlementEnabledAndNotInternal() throws Exception {
+        Stack stack = mock(Stack.class);
+        when(stack.getAccountId()).thenReturn(ACCOUNT_ID);
+        when(stack.getEnvironmentCrn()).thenReturn(ENVIRONMENT_CRN);
+        GatewayConfig gatewayConfig = mock(GatewayConfig.class);
+        when(stackService.getByIdWithListsInTransaction(1L)).thenReturn(stack);
+        Set<Node> nodes = Set.of(mock(Node.class));
+        when(freeIpaRecipeService.hasRecipeType(1L, RecipeType.POST_SERVICE_DEPLOYMENT, RecipeType.POST_CLUSTER_INSTALL)).thenReturn(false);
+        when(entitlementService.isWiamUsersyncEnabled(ACCOUNT_ID)).thenReturn(true);
+        FreeIpaClient ipaClient = mock(FreeIpaClient.class);
+        when(freeIpaClientFactory.getFreeIpaClientForStack(stack)).thenReturn(ipaClient);
+        when(ipaClient.findPermission("Set Password Expiration")).thenReturn(Set.of());
+        when(ipaClient.getUsernameLength()).thenReturn(5);
+        when(freeIpaClientFactory.getAdminUser()).thenReturn("adminka");
+        User user = new User();
+        user.setUid("adminid");
+        when(ipaClient.userFind("adminka")).thenReturn(Optional.of(user));
+
+        underTest.postInstallFreeIpa(1L, true);
+
+        verify(hostOrchestrator, never()).postServiceDeploymentRecipes(eq(gatewayConfig), eq(nodes), any(StackBasedExitCriteriaModel.class));
+        verify(userSyncBindUserService).createUserAndLdapConfig(stack);
+        verify(wiamClient).syncUsersInEnvironment(eq(ACCOUNT_ID), eq(ENVIRONMENT_CRN), anyString());
```

Comment:
nit: The call to wiam occurs in code after the permissions are set. I know that you aren't verifying InOrder, but it would be better to put the `verify(wiamClient)` and `verifyNoInteractions(usersyncService)` at the bottom of the test.